lib/spandex_phoenix.ex

```diff
@@ -239,7 +239,7 @@ defmodule SpandexPhoenix do
 
   defp mark_span_as_error(tracer, exception, stack) do
     # Normalize the span name for urls with no resource
-    if match?(%Phoenix.Router.NoRouteError{}, exception) do
+    if Code.ensure_loaded?(Phoenix.Router.NoRouteError) && Map.get(exception, :__struct__) == Phoenix.Router.NoRouteError do
```

Comment:
I'd like to make this a little simpler to make it clear that this is dependent on whether we have Phoenix included at all, as opposed to some optional part of Phoenix:

```suggestion
    if Code.ensure_loaded?(Phoenix) && Map.get(exception, :__struct__) == Phoenix.Router.NoRouteError do
```